### ISSUE TITLE
Plugin not compatible with Neos 5.X

### ISSUE THREAD

**Issue by mkuiphuis** [mocdk/MOC.NotFound#16]
This plugin does not appear to work anymore in Neos 5.X. Could have been caused by the introduction of PSR-7....

I tried to look into this issue myself, but couldn't get it working unfortunately. Any help would be greatly appreciated.

**Comment by timvonwerne** [comment]
Can confirm this.
The specific error message I get is:

> Undeclared arguments passed to ViewHelper MOC\NotFound\ViewHelpers\RequestViewHelper: path.

> Exception Code | 0
> -- | --
> Exception Type | TYPO3Fluid\Fluid\Core\ViewHelper\Exception
> Thrown in File | Packages/Libraries/typo3fluid/fluid/src/Core/ViewHelper/AbstractViewHelper.php
> Line | 476

**Comment by mkuiphuis** [comment]
Did actually find the solution for fixing the issue:
"The undeclared arguments passed issue could be due to the missing "initializeArguments() function in the ViewHelper. I have just added this snippet to the RequestViewHelper and I got my 404 page back:

```
/**
     * @return void
     * @throws \Neos\FluidAdaptor\Core\ViewHelper\Exception
     * @api
     */
    public function initializeArguments()
    {
        parent::initializeArguments();
        $this->registerArgument('path', 'string', 'Path');
    }
```

Haven't looked into the PSR-7 specifics though, so might not work for all use cases....

**Comment by aertmann** [comment]
@mkuiphuis I'm super busy these days, thanks for providing a fix. Would you be able to submit a pull request with your fix perhaps?

**Comment by aertmann** [comment]
Solved with #17 and https://github.com/mocdk/MOC.NotFound/releases/tag/3.1.3
